packages/theme/src/cli/commands/theme/dev.ts

```diff
@@ -121,20 +121,21 @@ export default class Dev extends ThemeCommand {
       controller.abort()
       controller = new AbortController()
       // eslint-disable-next-line @typescript-eslint/no-floating-promises
-      this.execute(adminSession, flags.password, command, controller)
+      this.execute(adminSession, flags.password, command, controller, true)
     }, this.ThemeRefreshTimeoutInMs)
 
     // eslint-disable-next-line @typescript-eslint/no-floating-promises
-    this.execute(adminSession, flags.password, command, controller)
+    this.execute(adminSession, flags.password, command, controller, false)
   }
 
   async execute(
     adminSession: AdminSession,
     password: string | undefined,
     command: string[],
     controller: AbortController,
+    shouldWait: boolean,
   ) {
-    await sleep(3)
+    if (shouldWait) await sleep(3)
```

Comment:
PS C:\Users\arkham\Desktop\shopify-apps\feb16-1745\extensions\prod-discount-ts> pnpm build

> prod-discount-ts@0.0.1 build C:\Users\arkham\Desktop\shopify-apps\feb16-1745\extensions\prod-discount-ts
> npm exec -- shopify app function build


╭─ success ────────────────────────────────────────────────────────────────────╮
│                                                                              │
│  Function built successfully.                                                │
│                                                                              │
╰──────────────────────────────────────────────────────────────────────────────╯

Reply:
oh I see, it's not the end of the world then. In theory we could use functions from https://github.com/unjs/get-port-please (which we already have as a dep) to check if the port is available for use :)

Reply:
Yeah, it was a temporary solution, the idea is that `dev` should handle token refresh internally, but that won't happen until the command is ported from Ruby...